### PR TITLE
feat(SCT-814): overhaul person view

### PR DESCRIPTION
**Pull request by jhackett1**
the first of a sequence of pull requests for a more mature, scalable and navigable person view: https://hackney.atlassian.net/browse/SCT-814

we did a lot of early user research on different approaches to the person view. this one was the winner

this one:

- splits the person view out into tabs (timeline, allocations, relationships, details)
- redesigns the old records table into a timeline view, which also shows information about in-progress submissions against that person
- populates the other tabs (i.e. everything other than timeline) with the existing components as-is. this looks a little repetitive right now but it's best to keep the size of this pr down

among the ux improvements, we:

- show a summary of a resident's circumstances right up by their name (birthdate, id, allocations)
- show the counts of relationships and allocations right there in the menu
- make it easier and more obvious to "add something new" to a case
- create a secondary action area near the main side nav, for other common actions (eg. "send an email/text message"?)
- synthesise together records/cases and submissions data into a cohesive timeline, to give users a clear idea of where a case is at

in later prs we'll:
- fix up the other tabs to make everything look more visually consistent and tightly integrated
- improve the way the "add something new" button works, with an improved hierarchy of case recording possibilities
- enhance the timeline further to show next steps and more filtering options

Screenshot 2021-07-13 at 11-32-24 Josephine Tester Social care Hackney Council
^ flavour of the new timeline view

### CODE REVIEW

File: components/NewPersonView/Layout.tsx
Comment (JohnFarrellDev) [inline components/NewPersonView/Layout.tsx:46]: I know the previous if statement would already have returned but I think saying `if (allocations?.length > 2)` would make more sense.

File: components/NewPersonView/PersonTimeline.tsx
Comment (JohnFarrellDev) [inline components/NewPersonView/PersonTimeline.tsx:12]: never used

File: components/NewPersonView/PersonTimeline.tsx
Comment (JohnFarrellDev) [inline components/NewPersonView/PersonTimeline.tsx:14]: What do we decide is minor vs major?

It would be more sensible to store this information in the root level data folder than in our component code.

File: components/NewPersonView/UnfinishedSubmissions.tsx
Comment (JohnFarrellDev) [inline components/NewPersonView/UnfinishedSubmissions.tsx:19]: It's typically better to use `Number.isNaN(...)` if you actually want to check for NaN value.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN

File: components/NewPersonView/UnfinishedSubmissions.tsx
Comment (JohnFarrellDev) [inline components/NewPersonView/UnfinishedSubmissions.tsx:22]: Would totalSteps ever be undefined, if so Number(undefined) gives us `NaN`,

File: pages/api/submissions/index.ts
Comment (JohnFarrellDev) [inline pages/api/submissions/index.ts:25]: I think this code is fine, there seems to be some issue on staging currently with getting unfinished submissions, should investigate that before merging this in.

File: pages/people/[id]/index.tsx
Comment (JohnFarrellDev) [inline pages/people/[id]/index.tsx:4]: `User` not being used

File: pages/people/[id]/index.tsx
Comment (JohnFarrellDev) [inline pages/people/[id]/index.tsx:12]: not being used

File: pages/people/[id]/index.tsx
Comment (JohnFarrellDev) [inline pages/people/[id]/index.tsx:66]: `isAuthorised` doesn't return a promise, we don't need to await right?

**Review by JohnFarrellDev — APPROVED** [review]
I think this is really good, just minor suggestions and let's see what is going on with not being able to get the unfinished submissions on staging

File: pages/people/[id]/index.tsx
Comment (chubberlisk) [inline pages/people/[id]/index.tsx:55]: What do you think about feature flagging this? So if it's on, use this improved person view and if it's not, use the old `PersonDetailsPage`. This is quite a big visual change so I wonder if we'll need approval before it goes out but tbh I think it's always good to put a feature flag just in case.

File: utils/api/submissions.spec.ts
Comment (chubberlisk) [inline utils/api/submissions.spec.ts:7]:
```suggestion
  describe('useUnfinishedSubmissions', () => {
```

File: utils/api/submissions.spec.ts
Comment (chubberlisk) [inline utils/api/submissions.spec.ts:8]: Looks like we're also doing some filtering on the response we get back. Could we add some unit tests around this?

File: utils/date.ts
Comment (chubberlisk) [inline utils/date.ts:79]: Could something like Moment.js help us here? https://momentjs.com/docs/#/parsing/special-formats/

**Review by chubberlisk — COMMENTED** [review]
Had a quick looksies as I know John has reviewed already. Tested locally and the new design is a nice breakdown of information! 💯 Some comments below.

File: components/NewPersonView/PersonTimeline.tsx
Comment (jhackett1) [reply components/NewPersonView/PersonTimeline.tsx:14]: yes, we should do that, but at the moment i don't have a good sense of what's major and minor - it will probably be a longer list than just these two.

File: pages/api/submissions/index.ts
Comment (jhackett1) [reply pages/api/submissions/index.ts:25]: was to do with api disruption - working fine now

File: utils/date.ts
Comment (jhackett1) [reply utils/date.ts:79]: we already have a date library in the app - moment is a very large library so would be best to avoid adding a second one. i think the ideal here is to fix the apis to always expose a standard iso format so we don't need this logic at all!

**Pull request merged**
